### DISCUSSION TITLE
Remeshing gives better results with doing index reduction

### DISCUSSION THREAD

**Discussion by fire** [zeux/meshoptimizer#229] [Ideas]
I noticed that if you re-mesh the mesh slightly by subdividing once and then distributing the vertices to be more even it causes the reduction to give better results.

Source http://www.gradientspace.com/tutorials/2020/1/2/libigl-in-unreal-engine.

Subdivision may require curl reduction so that the seams are merged but Directional is not robust. It causes infinite loops in my test inputs. https://avaxman.github.io/Directional/tutorial/

**Comment by zeux** [comment]
Thanks, that's good to know!

This isn't particularly surprising - as the simplifier is guided by topology, cleaner topology can result in cleaner output. It's also possible that starting from a more regularly sampled mesh results in better output as right now there's nothing in the simplifier that tries to ensure the quality of the output triangulation (e.g. avoiding skinny triangles).

I do have plans to experiment with remeshing in the future, but right now they're lower priority than other work, simplification-related or otherwise. It could be a valuable addition to the library depending on how it works, although remeshing in presence of arbitrary vertex attributes gets rather tricky so it's not clear if we'll go beyond retriangulation that can keep original vertices in tact...

Since I don't think there's anything actionable for this issue right now I'm going to close this (this probably works best as a discussion but unfortunately GitHub doesn't seem to support converting issues to discussion threads :(

**Comment by fire** [comment]
I had some good experiences with my fork of geometry3 https://github.com/fire/geometry3cpp.

My plan is to remesh to get a base mesh with sufficient even polygons and then run meshoptimizer for lod generation.

remesh is slow, but meshoptimizer is fast and robust! <3

**Reply by zeux** [reply]
Sounds good! fwiw I've converted the issue to a discussion since I realized GH actually does know how to do that, although I expected it to just morph the original issue instead of locking it.
